x-pack/plugins/infra/public/observability_logs/log_stream_query_state/src/url_state_storage_service.ts

```diff
@@ -90,23 +104,92 @@ export const initializeFromUrl =
   > =>
   (_context, _event) =>
   (send) => {
-    const queryValueFromUrl = urlStateStorage.get(filterStateKey) ?? defaultFilterStateValue;
+    const filterQueryValueFromUrl =
+      urlStateStorage.get(filterStateKey) ?? requiredDefaultFilterStateValue;
+    const filterQueryE = decodeFilterQueryValueFromUrl(filterQueryValueFromUrl);
 
-    const queryE = decodeQueryValueFromUrl(queryValueFromUrl);
+    // NOTE: Access logPosition for backwards compatibility with values previously stored under that key.
+    const positionQueryValueFromUrl = urlStateStorage.get(positionStateKey) ?? {};
```

Comment:
I wonder if there is a race condition between this and the initialization of the position state machine. :thinking: Does the position state machine remove the legacy keys that have been moved?

Reply:
Ah! Thank you for reminding me 🙈

When I originally modelled this the position machine was actually instantiated a level lower in the page state machine for exactly this reason. So it wasn't part of the received initial params flow. However, this made things read a little less nicely overall, and came with it's own issues. I ended up moving it to where it is now when I changed things to co-ordinate all state reading and event dispatching via the page state machine only, but I clearly forgot about this. There is 100% scope for a race condition. I'll make sure there's something explicit to stop the keys being wiped.